lib/graphlient/client.rb

```diff
@@ -51,11 +54,17 @@ def http(&block)
     end
 
     def schema
-      @schema ||= Graphlient::Schema.new(http, schema_path)
+      @schema ||= options[:schema] || Graphlient::Schema.new(http, schema_path)
     end
 
     private
 
+    def raise_error_if_invalid_configuration
+      if options[:schema_path] && options[:schema]
```

Comment:
Generally we do `.key?(:schema_path) && options.key?(:schema)`, you don't want to rely on the value, but on the fact that someone may have accidentally passed both in.